In [1]:
import pandas as pd
import numpy as np
import os

pd.set_option('max_colwidth', 200)
pd.set_option('display.width', 400)

from IPython.core.display import display, HTML
display(HTML("<style>.container { width:90% !important; }</style>"))

from pprint import pprint
from pathlib import Path

In [7]:
star_wars = pd.read_csv('./Data/StarWars.csv', sep=';')

In [13]:
star_wars.head()

respondentID seen  fan gender age_group             income                          loc                      yoda princess_leia_organa
0    3292879998   да   да      М     18-29                NaN           Южно-Атлантический            очень нравится       очень нравится
1    3292879538  нет  NaN      М     18-29         0 - 24.999     Юго-Западный центральный                       NaN                  NaN
2    3292765271   да  нет      М     18-29         0 - 24.999  Северо-Западный центральный  не знаю такого персонажа             нравится
3    3292763116   да   да      М     18-29  100.000 - 149.999  Северо-Западный центральный            очень нравится       очень нравится
4    3292731220   да   да      М     18-29  100.000 - 149.999  Северо-Западный центральный                  нравится             нравится

# Описание данных

- respondentID — идентификационный номер респондента
- seen — ответ на вопрос, видел ли респондент хотя бы один фильм из серии «Звездных войн» (да/нет)
- fan — ответ на вопрос, считает ли респондент себя фанатом серии фильмов (да/нет)
- gender — пол респондента (М/Ж)
- age_group — возрастная группа, к которой принадлежит респондент (18-29, 30-44, 45-60, > 60)
- income — доход респондента в долларах США (0 - 24.999, 25.000 - 49.999, 50.000 - 99.999, 100.000 - 149.999, 150.000+)
- loc — регион проживания респондента (в США)
- yoda — отношение респондента к персонажу серии (Йоде)
- princess_leia_organa — отношение респондента к персонажу серии (принцессе Лее Органе)

# Задания

#### К какой шкале относится переменная loc? Выберите один верный ответ.

In [11]:
star_wars['loc'].unique()

array(['Южно-Атлантический', 'Юго-Западный центральный',
       'Северо-Западный центральный', 'Средне-Атлантический',
       'Северо-Восточный центральный', 'Тихоокеанский', nan, 'Горный',
       'Новая Англия', 'Юго-Восточный центральный'], dtype=object)

#### Укажите, верно ли следующее утверждение: мужчины 18-29 лет, которые не являются фанатами серии, чаще всего проживают в Тихоокеанском регионе.

In [32]:
star_wars.loc[
    (star_wars['gender'] == 'М') &
    (star_wars['fan'] != 'да')
    ,
    'loc'
].mode()

0    Тихоокеанский
dtype: object

#### Какой регион проживания реже всего встречается в данных? Выберите один верный ответ.

In [59]:
pd.DataFrame(star_wars['loc'].value_counts()).sort_values('loc').head(1)

loc
Юго-Восточный центральный   38

#### Сколько респондентов не ответили на вопрос, нравится ли им Йода? Выберите один верный ответ.

In [50]:
star_wars[star_wars['yoda'].isna()].shape[0]

360

#### Скольким респондентам из Новой Англии очень нравится принцесса Лея Органа? Выберите один верный ответ.

In [56]:
star_wars.loc[
    (star_wars['loc'] == 'Новая Англия') &
    (star_wars['princess_leia_organa'] == 'очень нравится')
    ,:].shape[0]

39